ui/app/selectors/permissions.js

```diff
@@ -147,3 +147,55 @@ export function getPermissionDomainsMetadata (state) {
 function domainSelector (state, origin) {
   return origin && state.metamask?.domains[origin]
 }
+
+export function getAccountToConnectToActiveTab (state) {
+  const selectedAddress = getSelectedAddress(state)
+  const connectedAccounts = getConnectedAccountsForActiveTab(state)
+
+  const { metamask: { identities } } = state
+  const numberOfAccounts = Object.keys(identities).length
+
+  if (connectedAccounts.length && connectedAccounts.length !== numberOfAccounts) {
+    if (connectedAccounts.findIndex(({ address }) => address === selectedAddress) === -1) {
+      return identities[selectedAddress]
+    }
+  }
+
+  return undefined
+}
+
+export function getConnectedAccountsForActiveTab (state) {
+  const { activeTab, metamask } = state
+  const {
+    domains = {},
+    identities,
+    permissionsHistory,
+  } = metamask
+
+  const permissionsHistoryByAccount = permissionsHistory[activeTab.origin]?.['eth_accounts']?.accounts
+  const exposedAccounts = domains[activeTab.origin]?.permissions
+    ?.find(({ parentCapability }) => parentCapability === 'eth_accounts')
+    ?.caveats?.find(({ name }) => name === 'exposedAccounts').value || []
```

Comment:
This could be replaced with a call to `getPermittedAccountsForCurrentTab`, defined in this file:

```suggestion
  const exposedAccounts = getPermittedAccountsForCurrentTab(state)
```